app/code/Magento/InventoryInStorePickupFrontend/view/frontend/web/js/view/store-selector.js

```diff
@@ -0,0 +1,180 @@
+         */
+        setPickupInformation: function() {
+            var shippingAddress = quote.shippingAddress();
+
+            if (this.validatePickupInformation()) {
+                var sourceCode = _.findWhere(shippingAddress.customAttributes, {
+                    attribute_code: 'sourceCode',
+                });
+
+                shippingAddress = $.extend(true, quote.shippingAddress(), {
+                    extension_attributes: {
+                        pickup_location_code: sourceCode.value,
+                    },
+                    custom_attributes: {
+                        sourceCode: sourceCode.value,
+                    },
+                });
+
+                registry.async('checkoutProvider')(function(checkoutProvider) {
+                    checkoutProvider.set('shippingAddress', shippingAddress);
+
+                    setShippingInformationAction().done(function() {
+                        stepNavigator.next();
+                    });
+                });
+            }
+        },
+        /**
+         * @return {*}
+         */
+        getPopup: function() {
+            if (!this.popup) {
+                this.popup = modal(
+                    this.popUpList.options,
+                    $(this.popUpList.element)
+                );
+            }
+
+            return this.popup;
+        },
+        openPopup: function() {
+            var shippingAddress = quote.shippingAddress();
+
+            this.getPopup().openModal();
+
+            if (shippingAddress.city && shippingAddress.postcode) {
+                this.updateNearbyLocations(shippingAddress);
+            }
+        },
+        selectPickupLocation: function(location) {
+            pickupLocationsService.selectForShipping(location);
+            this.getPopup().closeModal();
+        },
+        isPickupLocationSelected: function(location) {
+            return _.isEqual(this.selectedLocation(), location);
+        },
+        updateNearbyLocations: function(address) {
+            var self = this;
+
+            return pickupLocationsService
+                .getNearbyLocations({
+                    radius: this.nearbySearchRadius,
+                    pageSize: this.nearbySearchLimit,
+                    country: this.defaultCountryId,
+                    city: address.city,
+                    postcode: address.postcode,
+                    region: address.region,
+                })
+                .then(function(locations) {
+                    self.nearbyLocations(locations);
+                })
+                .fail(function() {
+                    self.nearbyLocations([]);
+                });
+        },
+
+        validatePickupInformation: function() {
+            var emailValidationResult,
+                loginFormSelector = this.loginFormSelector;
+
+            if (!customer.isLoggedIn()) {
+                $(loginFormSelector).validation();
+                emailValidationResult = Boolean(
```

Comment:
Use ternary operation instead of creating boolean

Reply:
Done.